### ISSUE TITLE
Tcp keep alive setting function

### ISSUE THREAD

**Issue by testforvln** [zeromq/zmq.rs#155]
Hi all, I use this crate on receiving live stock quotes from stock exchanges. When the market daily rests for a while and then restarts trading, I could get no data from the zmq connection established before. 
It can be fixed by tcp keep alive settings, but I could not find the related function in this crate. Could you help me with it?

**Comment by poyea** [comment]
Hey there! I believe it isn't available as of [v0.3.3](https://github.com/zeromq/zmq.rs/releases/tag/v0.3.3) - as part of the option. Correct me with an example if I'm wrong. Ideally, we want to support something like this:
```rs
let mut options = SocketOptions::default();
// or a "TCP option"
options.tcp_keepalive(TcpKeepalive {
    keepalive: 1,
    count: 5,
    idle: 1,
    interval: 10,
});

let mut socket = zeromq::ReqSocket::with_options(options);
socket
    .connect("tcp://127.0.0.1:8888")
    .await
    .expect("Failed to connect");
```

**Comment by testforvln** [comment]
Yeah, this exmaple of with_options funciton is very helpful. I think it's the ideal way to implement the zmq options settings.

**Comment by paddor** [comment]
Sorry for hijacking. [OMQ.rs](https://github.com/paddor/omq.rs) implements TCP keepalive.
